File: shalvi-dsa-code/Python/superPow.py
from typing import List

class Solution:
    def superPow(self, a: int, b: List[int]) -> int:
        MOD = 1337

        def mod_pow(a: int, k: int) -> int:
            # Compute a^k % MOD
            a %= MOD
            result = 1
            for _ in range(k):
                result = (result * a) % MOD
            return result

        if not b:
            return 1

        last_digit = b.pop()
        part1 = mod_pow(self.superPow(a, b), 10)
        part2 = mod_pow(a, last_digit)

        return (part1 * part2) % MOD

File: shalvi-dsa-code/C/superPow.c
int modPow(int base, int exponent, int modulus) {
    int result = 1;
    base %= modulus;
    while (exponent > 0) {
        if (exponent % 2 == 1)
            result = (result * base) % modulus;
        exponent >>= 1;
        base = (base * base) % modulus;
    }
    return result;
}

int superPow(int a, int* b, int bSize) {
    int result = 1;
    for (int i = 0; i < bSize; ++i) {
        result = modPow(result, 10, 1337) * modPow(a, b[i], 1337) % 1337;
    }
    return result;
}